### PR TITLE
Lint edit actions

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

A picture is worth a thousand words:

screen shot 2017-11-10 at 6 13 48 pm

### CODE REVIEW

**Comment by rictic** [comment]
(this will fail until https://github.com/Polymer/polymer-linter/pull/122 is released)

File: src/commands/lint.ts [hunk 1]
```diff
@@ -61,6 +63,19 @@ export class LintCommand implements Command {
       type: Boolean,
       description: `Automatically fix as many issues as possible by ` +
           `updating your source on disk.`
+    },
+    {
+      name: 'edits',
+      type: String,
+      alias: 'e',
+      multiple: true,
+      description: `The lint edits to apply. Edits are usually less-safe fixes.`
```
Comment (valdrinkoshi) [inline src/commands/lint.ts]: Could you provide an example usage for `--edits`?

File: src/lint/lint.ts [hunk 2]
```diff
@@ -110,6 +114,18 @@ async function report(warnings: Warning[]) {
     if (infos.length > 0) {
       message += ` ${infos.length} ${chalk.green('info')} messages`;
     }
+    if (fixable > 0) {
+      if (editable > 0) {
+        message += `. ${fixable} can be automatically fixed with --fix ` +
+            `and ${editable} ${plural(editable, 'have', 'has')} edit actions`;
+      } else {
+        message += `. ${editable} ${plural(editable, 'have', 'has')} ` +
```
Comment (justinfagnani) [inline src/lint/lint.ts]: is this right?

File: src/lint/lint.ts [hunk 3]
```diff
@@ -179,9 +204,65 @@ function countEditsByFile(edits: Edit[]): ReadonlyMap<string, number> {
...
-function plural(n: number): string {
+function plural(n: number, pluralVal = 's', singularVal = ''): string {
   if (n === 1) {
-    return '';
+    return singularVal;
+  }
+  return pluralVal;
+}
+
+async function getEdits(
```
Comment (justinfagnani) [inline src/lint/lint.ts]: maybe add "prompt" into the name

File: src/lint/lint.ts [hunk 4]
```diff
@@ -179,9 +204,65 @@ function countEditsByFile(edits: Edit[]): ReadonlyMap<string, number> {
...
+      edits.push(warning.fix);
+    }
+    for (const action of warning.actions || []) {
+      if (action.kind === 'edit') {
+        if (editActionsToAlwaysApply.has(action.code)) {
+          edits.push(action.edit);
+          continue;
+        }
+        if (!options.noprompt) {
+          type ChoiceValue = 'skip'|'apply'|'apply-all';
```
Comment (justinfagnani) [inline src/lint/lint.ts]: for readability, consider extracting into a purely-prompting function

**Comment by rictic** [comment]
(this PR now blocked on https://github.com/Polymer/polymer-cli/pull/925 )

**Comment by rictic** [comment]
Looks like updating the analyzer may have made `build` unhappy. Tracking that down now

**Pull request merged**